### PR TITLE
Checkout: Make it possible to expose jQuery for paygate.js script

### PR DESCRIPTION
**Pull request by gziolo**
Another try to fix https://github.com/Automattic/wp-calypso/issues/3405.

It fixes the following issue:

> When I click the Pay button after filling the credit card details, a blue notice saying "Submitting payment" appears but nothing happens. I've tried several times and users are not charged, and nothing happens even after waiting 3-5 minutes.

This is a well known bug when using Electron:
https://github.com/atom/electron/issues/254

### Testing Desktop
1. Checkout `fix/3405-paygate-jquery-issue` branch in `calypso` folder.
2. Checkout `master` branch in main folder.
3. Make sure you have store sandbox enabled.
4. Execute `make run`.
5. Try to purchase any upgrade.
6. Provide data for new credit card to finalise purchase.
7. Purchase should be possible.

### Testing Calypso web
1. Checkout `fix/3405-paygate-jquery-issue` in wp-calypso
2. Execute `make run`.
3. Navigate to http://calypso.localhost:3000/
4. Try to purchase any upgrade.
5. Provide data for new credit card to finalise purchase.
6. Purchase should be possible.

### Testing Production build
1. From wp-calypso, run `CALYPSO_ENV=production make build`
2. Build should not print any error

### Reviews
- [x] Product Review
- [x] Code Review

/cc @peterbutler @Tug

### CODE REVIEW

File: client/lib/load-script/index.js [hunk 1]
```diff
@@ -45,6 +50,11 @@ var loadScript = function( url, callback ) {
 };
 
 var loadjQueryDependentScript = function( scriptURL, callback ) {
+	// It is not possible to expose jQuery globally in Electron App: https://github.com/atom/electron/issues/254.
+	// It needs to be loaded using require and npm package.
+	if ( config.isEnabled( 'desktop' ) ) {
+		window.jQuery = require( 'jquery' );
```
Comment (drewblaisdell) [inline client/lib/load-script/index.js]: Should we set `window.$` as well?

**Comment by drewblaisdell** [comment]
I'm getting `Uncaught ReferenceError: jquery is not defined` when I try to submit a purchase with new payment details on the desktop app:

screen shot 2016-02-26 at 11 09 51 am

File: client/lib/load-script/index.js (hunk 1 above)
Comment (Tug) [inline client/lib/load-script/index.js]: Yeah good idea!

**Comment by Tug** [comment]
Thanks for reporting this bug Drew, you're right I may have a stale version when I tested.
Could you try again with this change and if it doesn't work remove your build files and try again?:)

File: webpack.config.js [hunk 2]
```diff
@@ -88,6 +88,9 @@ if ( CALYPSO_ENV === 'desktop' || CALYPSO_ENV === 'desktop-mac-app-store' ) {
 	webpackConfig.entry.vendor = [ 'react', 'store', 'page', 'wpcom-unpublished', 'jed', 'debug' ];
 	webpackConfig.plugins.push( new webpack.optimize.CommonsChunkPlugin( 'vendor', '[name].[hash].js' ) );
 	webpackConfig.plugins.push( new ChunkFileNamePlugin() );
+	// jquery is only needed in the build for the desktop app
+	// see electron bug: https://github.com/atom/electron/issues/254
+	webpackConfig.externals.push( 'jquery' );
```
Comment (drewblaisdell) [inline webpack.config.js]: Why can't we define this as an external in the same array as `electron`?

**Comment by drewblaisdell** [comment]
It worked. :tada: 

I had one question about the code, but otherwise, it LGTM.

**Comment by scruffian** [comment]
@Tug, I added a commit to specify jQuery in the list of external dependencies as @drewblaisdell suggested. It seems to work fine like this. Was there a reason to do it the other way?

**Comment by Tug** [comment]
Well that's exactly how it was when @drewblaisdell tested and had the error. But I think it's not related, it seems the `public/build-desktop.js` file is not rebuilt when we change something in webpack's config!

**Comment by Tug** [comment]
Yep I can reproduce Drew's error with your last commit. @scruffian please clean your build files and try it ;)

Plus it makes sense, we want to exclude jQuery from the main build for non-desktop env only. For the desktop app we actually need this dependency when we require it there https://github.com/Automattic/wp-calypso/pull/3506/files#diff-7bb53ab371ab5975a08692c713885f4cR56

**Comment by scruffian** [comment]
Makes sense, thanks @Tug, I removed my commit.

**Comment by fabianapsimoes** [comment]
Product :+1:

**Pull request merged**